### ISSUE TITLE
How can I make sure that the gradient accumulation is actually added?

### ISSUE THREAD

**Issue by xiabingquan** [huggingface/accelerate#2518]
As the title suggests, I want to verify that the gradient accumulation is added. How can I do that?
I looked up the source code of `with accelerator.accumulate(model)`, but didn't find any insights.
https://github.com/huggingface/accelerate/blob/ee004674b9560976688e1a701b6d3650a09b2100/src/accelerate/accelerator.py#L1001-L1032

My main training loop is as follows:
```python
            with accelerator.accumulate(model):
                # forward
                res = model(data["feats"], data["feat_lens"], data["ys_in"])
                # weighted loss
                loss = some_loss_function(res)
                # backward
                optimizer.zero_grad()
                accelerator.backward(loss)
                optimizer.step()
                scheduler.step()
```

**Comment by muellerzr** [comment]
You want to look at things that happen when the `gradient_synchronization` is enabled in the GradientState. You can also see tests in the test_script.py and test_synchronization.py scripts to find our tests related to them
